gradio/components/video.py

```diff
@@ -199,7 +202,20 @@ def preprocess(self, payload: VideoData | None) -> str | None:
                 raise gr.Error(
                     f"Video is too long, and must be at most {self.max_length} seconds"
                 )
-
+        # TODO: Check other image extensions to see if they work.
+        valid_watermark_extensions = [".png", ".jpg", ".jpeg"]
+        if self.watermark_file is not None:
+            if not isinstance(self.watermark_file, (str, Path)):
+                raise gr.Error(
```

Comment:
By default, `gr.Error` are displayed to end users of the app, which should only be done if the error is a result of user error. Here, I'd say its probably the developer error, so better to keep it restricted to the error trace in the console:
```suggestion
                raise ValueError(
```

Reply:
I see, interesting. Thanks!